lib/rucio/web/ui/flask/main.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright 2014-2020 CERN
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+# - Thomas Beermann <thomas.beermann@cern.ch>, 2020
+
+
+from flask import Flask, request, render_template, make_response
+
+from rucio.api.authentication import get_auth_token_x509
+from rucio.common.config import config_get, config_get_bool
+from rucio.web.rest.utils import generate_http_error_flask
+from rucio.web.ui.flask.common.utils import get_token, authenticate, userpass_auth, x509token_auth, saml_auth, oidc_auth, finalize_auth, AUTH_ISSUERS, SAML_SUPPORT
+
+MULTI_VO = config_get_bool('common', 'multi_vo', raise_exception=False, default=False)
+POLICY = config_get('policy', 'permission')
+ATLAS_URLS = ()
+OTHER_URLS = ()
+
+application = Flask(__name__)
```

Comment:
since all the Flask endpoints are currently in one application in `lib/rucio/web/rest/flaskapi/v1/main.py`, it should be possible to register the ui as a blueprint there as well.

Actually registering it depends on #4043

Reply:
What would be the advantage?

Reply:
getting rid of the WSGI script aliases in the apache config. #4043 is coming anyway, so having to configure the ui endpoint specifically in the apache config just makes it harder and more confusing to set up.

Reply:
okay. In the webui case it's anyway only one alias pointing to '/' and it's usually not deployed together with the api. But I'll add it to be consistent with the api.

Reply:
ah, right, I was thinking of having the /ui endpoint for testing/demo purposes. Is that not supported?

Reply:
yes, that's the only real use case I can think of where you would have the rest api and webui running within the same apache. Like it's done in the dev container.